### PR TITLE
AEROGEAR-2172 Documentation available that explains how keyclok service works

### PR DESCRIPTION
**Pull request by vchepeli**
## Motivation
[AEROGEAR-2172](https://issues.jboss.org/browse/AEROGEAR-2172)
## Description
- Adding info how to provision keycloak service
- Adding info how to bind keycloak service

### CODE REVIEW

File: docs/walkthroughs/provision-keycloak-apb.adoc [hunk 1]
```diff
@@ -0,0 +1,96 @@
+[[provision-keycloak-service]]
+= Provision KeyCloak Service
+
+== Provision
+Choose the type of service you would like to provision:
+
+* Keycloak service
+* Linked Keycloak service
```
Comment (wei-lee) [inline docs/walkthroughs/provision-keycloak-apb.adoc]: I think `a new Keycloak service` and `link to an existing keycloak service` is easier to understand.

Comment (wei-lee) [reply docs/walkthroughs/provision-keycloak-apb.adoc]: is this the text that user will see in the provisioning wizard?

File: docs/walkthroughs/provision-keycloak-apb.adoc [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+NOTE: When the deprovision finishes, the Keycloak instance should be removed and it's web console should not be reachable.
+
+=== Linked KeyCloak service
+Deleting a linked Keycloak service will delete the following resources: 
+
+* client created in the specified realm
+* secrets
+* config maps
+
+NOTE: When the provision finishes, the realm created by provisioning the linked KeyCloak service should still exist
```
Comment (wei-lee) [inline docs/walkthroughs/provision-keycloak-apb.adoc]: should be `deprovision finishes`

File: docs/walkthroughs/provision-keycloak-apb.adoc (hunk 1 above)
Comment (maleck13) [inline docs/walkthroughs/provision-keycloak-apb.adoc]: I think this should be named ```shared keycloak service```

File: docs/walkthroughs/provision-keycloak-apb.adoc [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+Choose the type of service you would like to provision:
+
+* Keycloak service
+* Linked Keycloak service
+
+Go to the OpenShift web console. Open the *Mobile* Tab > *Services* from the service catalog and select the Keycloak service.
+This should open the Keycloak service provision wizard. Press the *Next* button and provide the configuration for the KeyCloak service
+
+=== KeyCloak service
+This service type can be shared between different Keycloak services.
```
Comment (maleck13) [inline docs/walkthroughs/provision-keycloak-apb.adoc]: I think we should say ```shared between different projects```?

File: docs/walkthroughs/provision-keycloak-apb.adoc [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+*The configuration would ask for the following parameters:*
```
Comment (maleck13) [inline docs/walkthroughs/provision-keycloak-apb.adoc]: The configuration would ask for the following parameters > The configuration will ask for the following parameters

File: docs/walkthroughs/provision-keycloak-apb.adoc [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+* *Add to Project(required)*: Create a new project or point to an existing project
+* *Keycloak admin username(required)*: The username of the admin user for the Keycloak web console
```
Comment (maleck13) [inline docs/walkthroughs/provision-keycloak-apb.adoc]: I think we should add: ```this user should be able to administer the realm or create new realms```

File: docs/walkthroughs/provision-keycloak-apb.adoc [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+* *Keycloak admin password(required)*: The password of the admin user for the Keycloak web console
+* *Connect to an existing shared service*: Remain unchecked
+* *URL of the shared service*: Remain empty
+* *Name of the Keycloak realm (defaults to current namespace)*: Remain empty
+
+After the service has been provisioned, you are then able to open the Keycloak web console
+with the address 'http://keycloak-<project-name>.<routing-suffix>' and the ```<project-name>``` realm should be available.
+
+=== Linked KeyCloak service
```
Comment (maleck13) [inline docs/walkthroughs/provision-keycloak-apb.adoc]: ```Linked``` > ```Shared```

File: docs/walkthroughs/provision-keycloak-apb.adoc [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+This service type is linked to an already existing KeyCloak instance.
+
+*The configuration would ask for the following parameters:*
```
Comment (maleck13) [inline docs/walkthroughs/provision-keycloak-apb.adoc]: Perhaps: ```The configuration of a shared service needs the following parameters```

File: docs/walkthroughs/provision-keycloak-apb.adoc [hunk 8]
```diff
@@ -1,22 +1,22 @@
...
 * Keycloak service
-* Linked Keycloak service
+* Shared Keycloak service
 
 Go to the OpenShift web console. Open the *Mobile* Tab > *Services* from the service catalog and select the Keycloak service.
 This should open the Keycloak service provision wizard. Press the *Next* button and provide the configuration for the KeyCloak service
 
 === KeyCloak service
-This service type can be shared between different Keycloak services.
+This service type can be shared between shared between different projects
```
Comment (dimitraz) [inline docs/walkthroughs/provision-keycloak-apb.adoc]: there's an extra 'shared between' here

File: docs/walkthroughs/provision-keycloak-apb.adoc [hunk 9]
```diff
@@ -25,17 +25,17 @@ This service type can be shared between different Keycloak services.
 After the service has been provisioned, you are then able to open the Keycloak web console
 with the address 'http://keycloak-<project-name>.<routing-suffix>' and the ```<project-name>``` realm should be available.
```
Comment (dimitraz) [inline docs/walkthroughs/provision-keycloak-apb.adoc]: can this address be put in back ticks rather?

File: docs/walkthroughs/provision-keycloak-apb.adoc [hunk 10]
```diff
@@ -1,22 +1,22 @@
...
 === KeyCloak service
-This service type can be shared between different Keycloak services.
+This service type can be shared between shared between different projects
 
-*The configuration would ask for the following parameters:*
+*The configuration will ask for the following parameters:*
 
 * *Add to Project(required)*: Create a new project or point to an existing project
-* *Keycloak admin username(required)*: The username of the admin user for the Keycloak web console
+* *Keycloak admin username(required)*: This user should be able to administer the realm or create new realms
```
Comment (dimitraz) [inline docs/walkthroughs/provision-keycloak-apb.adoc]: would it be possible to add a space before each opening bracket on all list points, just to make it look consistent?

File: docs/walkthroughs/provision-keycloak-apb.adoc [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+== Bind
+Bindings can be created for a provisioned Keycloak service, whether it's a shared one or not.
+
+* Go to the project overview where your Keycloak service is provisioned.
+* Select the Keycloak service under the *Provisioned Services* section. This should open up the service's details.
+* Click *Create Binding* under the *Bindings* section. This should bring up the wizard for bind creation.
+* Fill in the parameters:
+- *Service Name (Required)*: This is the name of the service the client is for. This value will be used for creating the `client
+id` and `name` in the Keycloak realm.
+- *Name of Keycloak realm*: This is the name of the Keycloak realm where you want the client
```
Comment (maleck13) [inline docs/walkthroughs/provision-keycloak-apb.adoc]: Based on our conversation on Friiday, I believe we are removing the realm from the bind option

File: docs/walkthroughs/provision-keycloak-apb.adoc (hunk 9 above)
Comment (JameelB) [reply docs/walkthroughs/provision-keycloak-apb.adoc]: For some reason if the address is within back ticks, it renders with a `;` a the end: ```http://keycloak-<project-name>.<routing-suffix>;```

**Pull request merged**